src/main/java/seedu/address/model/person/predicates/TagsContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,44 @@
+package seedu.address.model.person.predicates;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.model.person.Person;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Tests that a {@code Person}'s {@code Tags} matches any of the keywords given.
+ */
+public class TagsContainsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    public TagsContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        // all keywords must be present in tag
+        for (String keyword : keywords) {
+            boolean match = false;
+            for (Tag tag : person.getTags()) {
```

Comment:
Would something like this be better since the other predicates use `stream` as well?
```java
boolean hasMatch = person.getTags().stream()
                    .anyMatch(tag -> tag.getTagName().equalsIgnoreCase(keyword.toLowerCase()));
```

Reply:
honestly I tried doing it in streams initially but I'm not familiar with streams so it was not worth it haha

Reply:
Note: I personally think the `for` loop outside makes it a bit more readable. Although possible to entirely use `stream`s, it might be hard to understand what the code is conveying.